### ISSUE TITLE
2.6.0-beta2 analyzers fail under dotnet CLI build, incompatible with net45 / netstandard1.1

### ISSUE THREAD

**Issue by tillig** [dotnet/roslyn-analyzers#1480]
Using:
- dotnet 2.0.3 (from command line)
- Visual Studio 15.5.2
- Microsoft.CodeAnalysis.Analyzers 2.6.0-beta2
- Microsoft.CodeAnalysis.FxCopAnalyzers 2.6.0-beta2
- Microsoft.Net.RoslynDiagnostics 2.6.0-beta2

I'm working with the analyzers in the context of the [Autofac](https://github.com/autofac/Autofac) project. If you want to see what I'm seeing, you can play in there.

We were using version 2.3.0-beta1 of the above listed analyzers while on VS 15.3. With the recent release of 15.5.2 these analyzers stopped working, throwing errors. Following [the documentation here](https://github.com/dotnet/roslyn-analyzers/blob/master/README.md#recommended-version-of-analyzer-packages) I upgraded the `Autofac.csproj` (the only project with analyzers) to 2.6.0-beta2.

(Note you may want to update that doc - VS 15.5 isn't a preview anymore. Anyway...)

**When building in Visual Studio, this works fine.** No errors, no warnings.

**The challenge comes when we build at the command line**, which we do by calling `dotnet pack` from a command line (PowerShell) script. Doing that, we get this error something like 300 times (for different analyzers, but I grabbed the first one):

`CSC : error CS8032: An instance of analyzer Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.DiagnosticAnalyzerAttributeAnalyzer cannot be created from C:\Users\tillig\.nuget\packages\microsoft.codeanalysis.analyzers\2.6.0-beta2\analyzers\dotnet\cs\Microsoft.CodeAnalysis.Analyzers.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=2.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [C:\Users\tillig\Documents\Visual Studio 2017\Projects\Autofac\src\Autofac\Autofac.csproj]`

At first this looked like the #1425 issue - it can't find `Microsoft.CodeAnalysis`.

To try fixing it, I manually added a reference to `Microsoft.CodeAnalysis` to the project. The references ended up looking like this:

```xml
  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis" Version="2.6.0">
      <PrivateAssets>All</PrivateAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="2.6.0-beta2">
      <PrivateAssets>All</PrivateAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.0-beta2">
      <PrivateAssets>All</PrivateAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Net.RoslynDiagnostics" Version="2.6.0-beta2">
      <PrivateAssets>All</PrivateAssets>
    </PackageReference>
    <PackageReference Include="StyleCop.Analyzers" Version="1.1.0-beta004">
      <PrivateAssets>All</PrivateAssets>
    </PackageReference>
  </ItemGroup>
```

Note I used `Microsoft.CodeAnalysis` 2.6.0 - there's no `-beta2` version and I figured the release version would be better.

However, this brought on a new problem (trimmed out file paths for legibility):

```
error NU1202: Package Microsoft.CodeAnalysis.Workspaces.Common 2.6.0 is not compatible with net45 (.NETFramework,Version=v4.5). Package Microsoft.CodeAnalysis.Workspaces.Common 2.6.0 supports:
 - net46 (.NETFramework,Version=v4.6)
 - netstandard1.3 (.NETStandard,Version=v1.3)
```

Autofac targets `net45` and `netstandard1.1`. The previous analyzers allowed for that. Now they... don't? Seems like a breaking change if that was intentional (semver => 3.x rather than 2.x).

What is the magic combination of analyzers that will allow me to get Autofac building both from the dotnet CLI and Visual Studio?

**Comment by mavasani** [comment]
@tillig The issue seems to be that your project is using an older compiler toolset. You do not need to add a reference to Microsoft.CodeAnalysis NuGet package to change your toolset compiler, but instead need to update your dotnet CLI/SDK to the one that includes the 2.6.0 compilers.

**Comment by tillig** [comment]
OK, updating global.json to force dotnet 2.1.2 seems to have fixed it. That would probably be a good thing to add to the docs; a specific dotnet CLI/SDK version hasn't really been a thing in the past. The 2.3.0-beta1 versions ran fine under 1.1.0.

Thanks!

**Comment by mavasani** [comment]
> a specific dotnet CLI/SDK version hasn't really been a thing in the past

Sure, I will add the information to our docs so it is easy to find in future! Thanks.

**Comment by yoDon** [comment]
In case anyone else is having this issue in connection with Visual Studio 2017, I found that updating to the latest VS2017 (15.5.2 at the time I upgraded) fixed this issue for me.

**Comment by li0nsar3c00l** [comment]
Latest version of Visual Studio and .NET Core SDK and still got the problem.

**Comment by 333fred** [comment]
@li0nsar3c00l what version of the roslyn analyzers are you using?
